transports/http/src/main/java/org/mule/transport/http/HttpMessageProcessTemplate.java

```diff
@@ -192,13 +193,20 @@ private MessagingException getExceptionForCreatingFailureResponse(MessagingExcep
     @Override
     public void afterFailureProcessingFlow(MuleException exception) throws MuleException
     {
+        // MULE-11535: unwrap errors thrown in the initialization phase so that the true cause can be used to generate status codes
+        Throwable error = exception;
+        if (error instanceof DefaultMuleException && error.getCause() instanceof CreateException)
```

Comment:
Why is the DefaultMuleException check required? Isn't enough to do a null check on the causes and the CreateException one?